src/cargo/util/toml/targets.rs

```diff
@@ -752,7 +752,7 @@ fn infer_any(package_root: &Path, entry: &DirEntry) -> Option<(String, PathBuf)>
 
 fn infer_file(package_root: &Path, entry: &DirEntry) -> Option<(String, PathBuf)> {
     let path = entry.path();
-    let stem = path.file_stem()?.to_str()?.to_owned();
+    let stem = path.file_name()?.to_str()?.to_owned();
```

Comment:
If you could put the fix in a commit after the test is added, that way the change in the test will show what the fix did

Reply:
I separated the test and fix in separate commits, with the test failing in the first commit. Is that what you meant? Or do you want the test passing immediately, with changes to the test in the commit with the fix?

Reply:
Alright, will adjust.